ffi/src/utils.rs

```diff
@@ -9,27 +9,40 @@
 // specific language governing permissions and limitations under the License.
 
 pub mod strings {
-    use std::os::raw::c_char;
+    use std;
     use std::ffi::{
         CString,
         CStr
     };
+    use std::os::raw::c_char;
+    use std::rc::Rc;
+
+    use mentat::{
+        NamespacedKeyword,
+    };
 
     pub fn c_char_to_string(cchar: *const c_char) -> String {
         let c_str = unsafe { CStr::from_ptr(cchar) };
-        let r_str = match c_str.to_str() {
-            Err(_) => "",
-            Ok(string) => string,
-        };
+        let r_str = c_str.to_str().unwrap_or("");
         r_str.to_string()
     }
 
-    pub fn string_to_c_char(r_string: String) -> *mut c_char {
-        CString::new(r_string).unwrap().into_raw()
+    pub fn string_to_c_char<T>(r_string: T) -> *mut c_char where T: Into<String> {
+        CString::new(r_string.into()).unwrap().into_raw()
+    }
+
+    pub fn kw_from_string(mut keyword_string: String) -> NamespacedKeyword {
+        let attr_name = keyword_string.split_off(1);
+        let parts: Vec<&str> = attr_name.split("/").collect();
+        NamespacedKeyword::new(parts[0], parts[1])
     }
 
-    pub fn str_to_c_char(r_string: &str) -> *mut c_char {
-        string_to_c_char(r_string.to_string())
+    pub fn c_char_from_rc(rc_string: Rc<String>) -> *mut c_char {
+        if let Some(str_ptr) = unsafe { Rc::into_raw(rc_string).as_ref() } {
```

Comment:
Does this work instead?

```
rc.try_unwrap()                    // Get the value out of the `Rc` if we're the only owner.
  .map(CString::new)               // Make a CString.
  .map(|cs| cs.into_raw())         // Turn it into a raw pointer if successful.
  .unwrap_or_else(std::ptr::null_mut())   // Otherwise fail.
```

Reply:
nope. `try_unwrap` works if you are calling from Rust, but once you invoke this from Swift, `try_unwrap` fails. This was the _only_ way I could get the `String` out of the `Rc`, but I also figured there must be a better way.